core/model/src/main/java/org/eclipse/rdf4j/model/impl/NumericLiteral.java

```diff
@@ -113,4 +122,37 @@ public float floatValue() {
 	public double doubleValue() {
 		return number.doubleValue();
 	}
+
+	@Override
+	public String getLabel() {
+		return XMLDatatypeUtil.toString(number);
+	}
+
+	@Override
+	public Optional<String> getLanguage() {
+		return Optional.empty();
+	}
+
+	@Override
+	public IRI getDatatype() {
+		return datatype.getIri();
+	}
+
+	@Override
+	public CoreDatatype getCoreDatatype() {
+		return datatype;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o instanceof NumericLiteral) {
+			return datatype == ((NumericLiteral) o).datatype && number.equals(((NumericLiteral) o).number);
+		} else {
+			return super.equals(o);
+		}
+	}
```

Comment:
I cleaned up the equals method a bit. Since datatype is an enum it is faster to check than the number value, that's why I switched them around.

Reply:
Good for me.